aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -96,10 +83,18 @@ std::tuple<Tensor, Tensor> slogdet(const Tensor& self) {
   return std::make_tuple(det.sign(), diag_U.abs_().log_().sum());
 }
 
+Tensor pinverse(const Tensor& self) {
+  AT_CHECK(at::isFloatingType(self.type().scalarType()) && self.dim() == 2,
+           "pinverse(", self.type(), "{", self.sizes(), "}): expected a 2D tensor "
+           "of floating types");
+  Tensor U, S, V;
+  std::tie(U, S, V) = self.svd();
+  Tensor S_pseudoinv = at::where(S != 0.0, S.reciprocal(), at::zeros({}, self.options()));
```

Comment:
Should this be > 1e-06 or something? I think the singular values might be a reason for failing tests.

Reply:
Let's do the [numpy](https://docs.scipy.org/doc/numpy/reference/generated/numpy.linalg.pinv.html) thing and introduce another `rcond` parameter s.t. `s[i] = s[i] > rcond * s[0] ? s[i] : 0`.

Btw, why did you call this `pinverse` and not `pinv`?

Reply:
Let's also add an entry in `test_autograd`. Since this uses `svd` it might be tricky as some cases will fail `gradgradcheck` due to `nan`. You can check the `svd` test cases, and use them as a reference.

Reply:
The MATLAB rule seems interesting too: https://www.mathworks.com/help/matlab/ref/pinv.html . But I think we should align with NP.